addons/web_editor/controllers/main.py

```diff
@@ -530,6 +530,16 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
             fields['res_id'] = res_id
         if name:
             fields['name'] = name
+
+        # Restricted editors can handle attachments related to records to which they have access
+        # Would user be able to read fields of original record ?
+        if attachment and attachment.res_model and attachment.res_id:
+            request.env[attachment.res_model].browse(attachment.res_id).with_user(request.env.user).check_access_rights('read')
```

Comment:
Why only checking for `read` ? If a user can `read` a product record but can't `write` on a product record, why we should allow him/her to modify the related image ?

Reply:
Write access is checked on the target record on the next line, because that record is the one modified if a new image is written. Creating a modified image on another record does not modify the image stored in the origin record, this is therefore not a write access on the origin record.